src/RestApi/StoreApi/Utilities/ProductQuery.php

```diff
@@ -0,0 +1,351 @@
+<?php
+/**
+ * Helper class to handle product queries for the API.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Utilities;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Product Query class.
+ */
+class ProductQuery {
+	/**
+	 * Prepare query args to pass to WP_Query for a REST API request.
+	 *
+	 * @param \WP_REST_Request $request Request data.
+	 * @return array
+	 */
+	public function prepare_objects_query( $request ) {
```

Comment:
Doesn't need done in this pull, but would be great to break this method up a bit more into individual query part prep.  I think it'd make it more readable and easier to refactor if needed down the road.  Any place you've left comments (which are great btw!) are potential things to break out.

Reply:
Agree, this was copied from elsewhere. I'll leave for future refactor.